src/app/batch_generate_keys/batch_generate_keys.ml

```diff
@@ -0,0 +1,345 @@
+          $amount: UInt64!,
+          $token: UInt64,
+          $fee: UInt64!,
+          $nonce: UInt32,
+          $memo: String,
+          $field: String,
+          $scalar: String) {
+  sendPayment(input:
+    {from: $sender, to: $receiver, amount: $amount, token: $token, fee: $fee, nonce: $nonce, memo: $memo},
+    signature: {field: $field, scalar: $scalar}) {
+    payment {
+      id
+    }
+  }
+}
+|}]
+
+let make_graphql_signed_transaction ~sender_priv_key ~receiver ~amount ~fee =
+  let sender_pub_key =
+    Public_key.of_private_key_exn sender_priv_key |> Public_key.compress
+  in
+  let receiver_pub_key = Public_key.Compressed.of_base58_check_exn receiver in
+  let field, scalar =
+    Mina_base.Signed_command.sign_payload sender_priv_key
+      { common =
+          { fee
+          ; fee_token = Mina_base.Token_id.default
+          ; fee_payer_pk = sender_pub_key
+          ; nonce = Mina_numbers.Account_nonce.zero
+          ; valid_until = Mina_numbers.Global_slot.max_value
+          ; memo = Mina_base.Signed_command_memo.empty
+          }
+      ; body =
+          Payment
+            { source_pk = sender_pub_key
+            ; receiver_pk = receiver_pub_key
+            ; token_id = Mina_base.Token_id.default
+            ; amount
+            }
+      }
+  in
+  let graphql_query =
+    Send_payment.make
+      ~receiver:(Graphql_lib.Encoders.public_key receiver_pub_key)
+      ~sender:(Graphql_lib.Encoders.public_key sender_pub_key)
+      ~amount:(Graphql_lib.Encoders.amount amount)
+      ~fee:(Graphql_lib.Encoders.fee fee)
+      ~field:(Snark_params.Tick.Field.to_string field)
+      ~scalar:(Snark_params.Tick.Inner_curve.Scalar.to_string scalar)
+      ()
+  in
+  let graphql_query_json =
+    `Assoc
+      [ ("query", `String graphql_query#query)
+      ; ("variables", graphql_query#variables)
+      ]
+  in
+  Format.sprintf
+    "curl 'http://127.0.0.1:3085/graphql' -X POST -H 'content-type: \
+     application/json' --data '%s'@."
+    (Yojson.Basic.to_string graphql_query_json)
+
+let there_and_back_again ~num_accts ~num_txn_per_acct ~txns_per_block ~slot_time
+    ~fill_rate ~rate_limit ~rate_limit_level ~rate_limit_interval
+    ~origin_sender_public_key_option ~origin_sender_secret_key_path_option () =
+  let rate_limit =
+    if rate_limit then
+      let slot_limit =
+        Float.(
+          of_int txns_per_block /. of_int slot_time *. fill_rate
+          *. of_int rate_limit_interval)
+      in
+      let limit = min (Float.to_int slot_limit) rate_limit_level in
+      Some limit
+    else None
+  in
+  let batch_count = ref 0 in
+  let base_send_amount = Currency.Amount.of_formatted_string "0" in
+  let fee_amount = Mina_base.Signed_command.minimum_fee in
+  let acct_creation_fee = Currency.Amount.of_formatted_string "1" in
+  let initial_send_amount =
+    (* min_fee*num_txn_per_accts + base_send_amount*num_txn_per_accts + acct_creation_fee *)
+    let total_send_value =
```

Comment:
Nit: this might be cleaner with `let open Option.Let_syntax in` and then just a final `Option.value_exn` at the end.